src/Acuminator/Acuminator.Utilities/DiagnosticSuppression/SuppressionManagerInitInfo.cs

```diff
@@ -0,0 +1,44 @@
+﻿using Acuminator.Utilities.Common;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Xml.Linq;
+
+namespace Acuminator.Utilities.DiagnosticSuppression
+{
+	public readonly struct SuppressionManagerInitInfo : IEquatable<SuppressionManagerInitInfo>
+	{
+		public string Path { get; }
+
+		public bool GenerateSuppressionBase { get; }
+
+		public SuppressionManagerInitInfo(string path, bool generateSuppressionBase)
+		{
+			path.ThrowOnNullOrWhiteSpace(nameof(path));
+
+			Path = path;
+			GenerateSuppressionBase = generateSuppressionBase;
+		}
+
+		public override bool Equals(object obj) => obj is SuppressionManagerInitInfo initInfo
+			? Equals(initInfo)
+			: false;
+
+		public bool Equals(SuppressionManagerInitInfo other) => 
+			GenerateSuppressionBase == other.GenerateSuppressionBase && 
+			string.Equals(Path, other.Path, StringComparison.OrdinalIgnoreCase);
+
+		public override int GetHashCode()
+		{
+			int hash = 17;
+
+			unchecked
```

Comment:
Unchecked is not needed here as we do not specify checked context in the project
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/checked-and-unchecked

Reply:
@metallmonster it is better to explicitly define it because we heavily rely on this behavior. It is a common pattern in C#.

Reply:
I don't see any reason to keep it: if a lot of people do it, it doesn't become useful); we rely on this behavior and it is OK as we have no reasons to specify checked context now or in the future.
But I don't mind if you want to keep it.

Reply:
I know that unchecked context is turned on by default. 
Still imho not specifying it explicitly is not a good thing:
1) The default value for this setting may change and the code should not break. The event has a low probability but still it is not impossible.
2) This is one of recommended implementations:
https://stackoverflow.com/questions/263400/what-is-the-best-algorithm-for-an-overridden-system-object-gethashcode
I agree with the author. Imho it is better to explicitly tell that overflow is expected behavior for the hashcode calculation than rely on the default mode